ultrasonic-pid/robot.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) FIRST and other WPILib contributors.
+# Open Source Software; you can modify and/or share it under the terms of
+# the WPILib BSD license file in the root directory of this project.
+#
+
+import wpilib
+import wpilib.drive
+import wpimath.controller
+import wpimath.filter
+
+
+class MyRobot(wpilib.TimedRobot):
+    """
+    This is a sample program to demonstrate the use of a PIDController with an ultrasonic sensor to
+    reach and maintain a set distance from an object.
+    """
+    
+    # distance the robot wants to stay from an object
+    # (one meter)
+    kHoldDistanceMillimeters = 1.0e3
+
+    # proportional speed constant
+    # negative because applying positive voltage will bring us closer to the target
+    kP = -0.001
```

Comment:
I'd rather inline the constants with comments or better yet use keyword arguments.

ie. `wpilib.PWMSparkMax(channel=0)`

Reply:
None of our other examples currently do that.

Reply:
I think that style is a perfectly fine choice (and indeed, desirable sometimes!), but one of the unstated goals of our examples is to provide a way for users to compare languages -- when you look at the python and java examples it should be obvious how they're the same and how they're different. By putting our constants inline I think it would be more challenging to see the similarities.

Reply:
:(